homeassistant/components/mqtt/vacuum/__init__.py

```diff
@@ -0,0 +1,97 @@
+"""
+Support for MQTT vacuums.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/vacuum.mqtt/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.vacuum import DOMAIN
+from homeassistant.components.mqtt import ATTR_DISCOVERY_HASH
+from homeassistant.components.mqtt.discovery import (
+    MQTT_DISCOVERY_NEW, clear_discovery_hash)
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['mqtt']
+
+CONF_COMPONENT = 'component'
```

Comment:
will `schema` be better?

Reply:
It started out as `schema` then I tricked @pszafer to change to `component`..
I think either is fine really. It is a different MQTT schema, and we use `schema` configuration variable in MQTT Light, but at the same time with the other schema a different component `StateVacuumDevice` will be instantiated instead of a `VacuumDevice`.

Reply:
I think we should follow mqtt light and call it schema.

Reply:
Renamed.